src/karma-webpack.js

```diff
@@ -383,9 +387,6 @@ function createPreprocesor(/* config.basePath */ basePath, webpackPlugin) {
         throw err;
       }
 
-      const outputPath = webpackPlugin.outputs.get(filename);
-      file.path = path.join(basePath, outputPath);
```

Comment:
How comes you removed this? It's needed to rewrite the asset path for karma to point to `.js` if the file extension was `.ts` for example

Reply:
okay, but with this, just for simple js file, the karma web server will not find the file and throw 404

Reply:
I tried to rewrite the file path like the above, but failed, so what do you think here should to do?

Reply:
Hi @rynclark , I tried to add this code back when testing with `.ts` file, but there's a problem, here `path.join(basePath, outputPath)` refers to the absolute path in the file system, so that 404, because `.js` doesn't exist at this `basePath`, so when replacing with path same as`webpackOptions.output.path = path.join(os.tmpdir(), '_karma_webpack_', indexPath, '/');`, still cannot find the compiled `.js` file

Reply:
Could we move discussion about this into another PR and first of all fix the file path normalization issue as this seems to be critical?

Reply:
@michael-ciniawsky, but as my test, for this part, even just fix the file path normalization issue, still cannot work with 404 error.

Reply:
I see, can't confidently comment on what went wrong here, we may need to revert this for now then... :)

If you remove the `this.outputs` part it works, so other changes made recently e.g `this.entries` are not affected or is there a general problem which this approach?

Reply:
This also isn't possibly related to #338 ?

Reply:
Right now, the current changes in this PR can work for me for `.js` file, but as @rynclark said, if it's for `.ts` file, this change cannot work. So as you can see the comment above, I tried to add this back, but there's a issue that I haven't found a solution yet.
I will try with #338 later.

Reply:
Right now, this can work for both `.js` and `.ts` based on my test. The only different with #353, is L393, add normalize again.

Reply:
I tried with v3.0.3, still needs this L393 changes which normalize the path again. I have opened a PR #354  for 3.0. @michael-ciniawsky

Reply:
@Teamop Thx. Do the changes introduced by https://github.com/webpack-contrib/karma-webpack/commit/ab4dde9a7b859e815e11925a0f5cc5c986360264 still cause any regressions (for you)? If so I may need to revert them for now. 

This PR is ready I assume ? :)

Reply:
this PR is ready, with this changs, I have no issues.

Reply:
Sry 🙏 I skipped over https://github.com/webpack-contrib/karma-webpack/pull/351/files#r215968728

Reply:
np, thanks @michael-ciniawsky